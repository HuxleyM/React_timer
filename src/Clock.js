import React, {Component} from 'react';
import TimeNow from './TimeNow'

class Clock extends Component{

    render(){
        const { mode } = this.props;

        return (
            <div>
                <h1> {(mode == 'clock')? 'The Time Is:' : 'Set Timer :'} </h1>
                { mode == 'clock' && < TimeNow />}
            </div>
        )
    }

}



export default Clock