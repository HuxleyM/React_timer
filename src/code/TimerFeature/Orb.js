import React, { Component } from 'react';


class Orb extends Component {

    constructor(props){
        super()
        this.state = {
            startTime : props.startTime,
            totalTime : props.endTime,
            totalHeight : window.innerHeight,
        }
    }

    difference(){
        return (this.state.totalTime  - this.state.startTime);
    
    }

    move(){
        let timeNow = Date.now(); 
        let differenceFromStart = timeNow - this.state.startTime   
        let overallDifference = this.difference();  

        let move = (differenceFromStart / overallDifference) * 100
        console.log(move)
        document.getElementById('orb').style.top=`${move}%`
    }

    render(){
        return <div id='orb'></div>
    }
}

export default Orb

