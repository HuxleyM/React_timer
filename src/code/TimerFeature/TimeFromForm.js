class TimeFromForm{
    constructor(endTime){
      this.from = Date.now();
      this.till = this.convertToUTCTime(endTime)
    }
 
    get timeFrom(){
      return this.from
    }
 
    get timeTill(){
     return this.till
    }

  convertToUTCTime(formTime){
    var endTime = new Date();
    let hours = endTime.setHours(this.get_hours(formTime))
    let mins = endTime.setMinutes(this.get_mins(formTime))
    let sec = endTime.setSeconds(0)
    return endTime.getTime()
  }

   get_hours(string){
     return parseInt(string.split('').splice(0,2).join(''))
   }
   
   get_mins(string){
       return parseInt(string.split('').splice(3,2).join(''))
   }
}

export default TimeFromForm