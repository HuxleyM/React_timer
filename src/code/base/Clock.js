import React from 'react';
import TimeNow from '../TimeNowFeature/TimeNow'
import Timer from '../TimerFeature/Timer'

const Clock = ({ mode }) => (
    <div>
        <h1> {(mode === 'clock')? 'The Time Is:' : 'Set Timer :'} </h1>
        { mode === 'clock' && < TimeNow />}
        { mode === 'timer' && < Timer />}
    </div>
)
    
export default Clock