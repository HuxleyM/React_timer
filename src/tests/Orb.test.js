import React from 'react';
import Clock from '../code/TimerFeature/TimeFromForm'
import {shallow, mount} from 'enzyme'
import renderer from 'react-test-renderer';
import TimeFromForm from '../code/TimerFeature/TimeFromForm';
import Orb from '../code/TimerFeature/Orb.js'


describe('orb component', ()=>{
    
    global.innerHeight = 640;
    const startTime = new Date('December 17, 2019 13:24:00').getTime();
    const endTime = new Date('December 17, 2019 14:24:00').getTime();
    const orbDummy = mount(<Orb endTime={endTime} change={startTime} />)



    it('constructs with a totaltime and totalHeight', ()=>{
        expect(orbDummy.state('totalTime')).toEqual(endTime)
        expect(orbDummy.state('totalHeight')).toEqual(640)
    })

    it('a call to move should move the orb', ()=>{
        var theOrb = mount(<Orb endTime={endTime} change={startTime} />)
        renderer.create(theOrb).toJSON();
        const midTime = new Date('December 17, 2019 13:54:00').getTime();
        theOrb.instance().move(midTime)
        const updatedTree = renderer.create(theOrb).toJSON();
        expect(updatedTree).toMatchSnapshot();
    })

    // ok so this error is happening because the div doesnt exist at this point in time
})

