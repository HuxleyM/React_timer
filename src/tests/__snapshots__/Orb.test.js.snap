// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`orb component a call to move should move the orb 1`] = `
<div
  id="orb"
/>
`;
